internal/run/convert_test.go

```diff
@@ -40,3 +47,127 @@ func TestV3ReleaseFound(t *testing.T) {
 	assert.True(t, v3ReleaseFound("myapp", cfg))
 	assert.False(t, v3ReleaseFound("doesnt_exists", cfg))
 }
+
+func clientsetWithV2ConfigmapsMock() *fake.Clientset {
+	return fake.NewSimpleClientset(
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "myapp.v1",
+				Namespace: "example",
+				Labels: map[string]string{
+					"NAME":    "myapp",
+					"OWNER":   "TILLER",
+					"STATUS":  "DEPLOYED",
+					"VERSION": "1",
+				},
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "myapp.v2",
+				Namespace: "example",
+				Labels: map[string]string{
+					"NAME":    "myapp",
+					"OWNER":   "TILLER",
+					"STATUS":  "DEPLOYED",
+					"VERSION": "2",
+				},
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "other.v1",
+				Namespace: "example",
+				Labels: map[string]string{
+					"NAME":    "other",
+					"OWNER":   "TILLER",
+					"STATUS":  "DEPLOYED",
+					"VERSION": "1",
+				},
+			},
+		},
+	)
+}
+
+func TestGetV2ReleaseConfigmaps(t *testing.T) {
+
+	c := NewConvert(env.Config{Release: "myapp", TillerNS: "example"}, "", "")
+	clientset := clientsetWithV2ConfigmapsMock()
+
+	cmaps, err := c.getV2ReleaseConfigmaps(clientset)
+	assert.NoError(t, err)
+	assert.Equal(t, len(cmaps.Items), 2)
+}
+
+func TestPreserveV2ReleaseConfigmaps(t *testing.T) {
+
+	c := NewConvert(env.Config{Release: "myapp", TillerNS: "example"}, "", "")
+	clientset := clientsetWithV2ConfigmapsMock()
+
+	clientset.Fake.AddReactor("update", "configmap", func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
```

Comment:
What is this reaction function doing? Further, I removed this and the test still passes, so is it necessary?

Reply:
I think this is necessary to mock the UPDATE calls in kubernetes fake client. Without this, when trying to retrieve the configmaps, even after running `c.preserveV2ReleaseConfigmaps`, the configmaps don't change. So this reactor is needed to return the object state when an "update" call is made

Reply:
I can also confirm, if I remove the reactor the tests continue to pass.  A reactor is a construct in the fakeclient that are used to change the behavior of the client when an action occurs.  In this case the reactor is just returning the same state of the configmaps.

Reply:
removed, thx